src/Microsoft.Dnx.Tooling/Restore/NuGet/NuGetv3Feed.cs

```diff
@@ -256,8 +262,9 @@ private static void EnsureValidFindPackagesResponse(Stream stream, string uri)
                 var json = JObject.Load(new JsonTextReader(new StreamReader(stream)));
                 if (json["versions"] == null)
```

Comment:
Maybe avoid doing multiple lookups for the same key?

```C#
var versions = json["versions'];
if (versions != null && versions.Type != JTokenType.Array)
{
   throw new InvalidDataException(
       $"{message} The value of 'versions' property is not an array.");
}
```

Maybe a future refactoring, but it would be useful to not reparse the json multiple times over (once here and once in the body of `FindPackagesByIdAsyncCore`). Perhaps this could stash the parsed json somewhere that flows via `data`?